dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionlet.java

```diff
@@ -110,12 +111,11 @@ public void executeAction(final WorkflowProcessor processor,
 					contentletDependencies.isRespectAnonymousPermissions() : user.isFrontendUser();
 
 			//Keeps existing relationships
-			if (contentletRelationships == null){
-                contentletRelationships = contentletAPI.getAllRelationships(contentlet);
-            }
+
 
 			if (categories == null) {
 				categories = this.categoryAPI.getParents(contentlet, user, respectFrontendPermission);
+				categories = UtilMethods.isSet(categories) ? categories : null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5397063)